packages/opentelemetry-exporter-datadog/src/transform.ts

```diff
@@ -96,56 +93,106 @@ function createSpan(
   ddSpanBaseContext._spanId = ddSpanId;
   ddSpanBaseContext._parentId = ddParentId;
 
-  // set reserved service and resource tags
-  ddSpanBase.addTags({
-    'resource.name': createResource(span),
-    'service.name': service_name,
-  });
-
   // set error code
+  addErrors(ddSpanBase, span);
+
+  // set span kind
+  addSpanKind(ddSpanBase, span);
+
+  // set span type
+  addSpanType(ddSpanBase, span);
+
+  // set datadog specific env and version tags
+  addDatadogTags(ddSpanBase, span, serviceName, env, version);
+
+  // set sampling rate
+  setSamplingRate(ddSpanBase, span);
+
+  // set span duration
+  setDuration(ddSpanBase, span);
+
+  // mark as finished span so that it is exported
+  ddSpanBaseContext._isFinished = true;
+
+  return ddSpanBase;
+}
+
+function addErrors(ddSpanBase: typeof Span, span: ReadableSpan): void {
   if (span.status && span.status.code && span.status.code > 0) {
     // TODO: set error.msg error.type error.stack based on error events
     // Opentelemetry-js has not yet implemented https://github.com/open-telemetry/opentelemetry-specification/pull/697
     // the type and stacktrace are not officially recorded. Until this implemented,
     // we can infer a type by using the status code and also the non spec `<library>.error_name` attribute
     const possibleType = inferErrorType(span);
-    ddSpanBase.setTag('error', 1);
-    ddSpanBase.setTag('error.msg', span.status.message);
+    ddSpanBase.setTag(DatadogDefaults.ERROR_TAG, 1);
```

Comment:
What does this `1` mean, can this have a different values? maybe add some explanation or soem enum what it really means - I see this in few other places so this might make sense to put here some const default

Reply:
the `error` span tag can be either 1 or 0 if set. 0 is ok, 1 is error. Generally it only gets set in the case of an error. I've moved this into a more descriptive enum with a code comment to clarify.